src/xdist/workermanage.py

```diff
@@ -57,8 +57,17 @@ def __init__(
         if self.testrunuid is None:
             self.testrunuid = uuid.uuid4().hex
         self.group = execnet.Group(execmodel="main_thread_only")
+        for proxy_spec in self._getpxspecs():
```

Comment:
I need to crosscheck in execnet on how proxy startup is managed

I vaguely recall that main thread only may cause trouble when more than one worker is behind a proxy

Reply:
I tested locally using 5 workers behind a local socket proxy, and it worked.

Reply:
I realized my mistake 

Proxy doesn't get thr default value of  main thread only as its using a own spec

Thanks for validation